### PR TITLE
Flag edit double check

### PR DESCRIPTION
**Pull request by rbatistadev**

### CODE REVIEW

File: library/src/components/shared/ConfirmEditDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    setProgress(TOTAL_TIME);
+
+    return () => {
+      if (timer) {
+        clearInterval(timer);
+      }
+    };
+  }, [open]);
```
Comment (danigargar) [inline library/src/components/shared/ConfirmEditDialog.tsx:59]:
```suggestion
  useEffect(() => {
    let timer: NodeJS.Timeout | null = null;
    if (open) {
      timer = setInterval(() => {
        setProgress((oldProgress) => {
          if (oldProgress === 0) {
            return 0;
          }
          return oldProgress - 5;
        });
      }, 250);
    }

    setProgress(TOTAL_TIME);

    return () => {
      if (timer) {
        clearInterval(timer);
      }
    };
  }, [open, TOTAL_TIME]);
```
The behavior will be exactly the same and this is what is recommended.

File: library/src/entities/DefaultEntityBehavior/Form/Form.tsx [hunk 2]
```diff
@@ -196,16 +222,42 @@ const Form: EntityFormType = (props) => {
...
     <div>
       <form
         onSubmit={(e) => {
+          e.preventDefault();
+          e.stopPropagation();
+
           if (formik.isSubmitting) {
-            e.preventDefault();
-            e.stopPropagation();
             return false;
```
Comment (danigargar) [inline library/src/entities/DefaultEntityBehavior/Form/Form.tsx:252]: returning false from onSubmit() will have any effect?

**Review by danigargar — CHANGES_REQUESTED** [review]

File: library/src/components/shared/ConfirmEditDialog.tsx (hunk 1 above)
Comment (rbatistadev) [reply library/src/components/shared/ConfirmEditDialog.tsx:59]: TOTAL_TIME is a constant, so, it doesn't have sense to check the possible changes on it

Comment (danigargar) [reply library/src/components/shared/ConfirmEditDialog.tsx:59]: right

**Review by danigargar — APPROVED** [review]

**Pull request merged**